### DISCUSSION TITLE
Tensorflow 2.16 support

### DISCUSSION THREAD

**Discussion by jmmanso** [openvinotoolkit/nncf#3030] [General] [answered]
Hi, when can we expect NNCF to support TF 2.16? It's an important version since it comes with Keras 3. Thanks

**Comment by MaximProshin** [comment]
Hi @jmmanso ! There is no plan yet as we don't see a wide adoption of TF path in NNCF. Can you please share some information on whether you use it or plan to use it and how?
cc @AlexKoff88

**Reply by jmmanso** [reply]
Hi Maxim. We're [Planet](https://www.planet.com), a large provider of satellite imagery and derived data, and use TF + NNCF extensively in production. Most of our computer vision models are small CNNs and we run them on Intel CPUs over large pools of nodes. We use NNCF to compile and quantize our TF models, and it has been more effective than any other approach we've tried, reducing our inference costs very significantly. Thus, if NNCF stops supporting TF, that would be a problem for us. 
Thanks

**Reply by MaximProshin** [reply]
Thanks for your reply! Did I get it right that you specifically use the Quantization-Aware Training of NNCF for TF models and then convert such models to OpenVINO? Just want to better understand your pipeline.

**Reply by jmmanso** [reply]
we actually just use post-training quantization, mostly the `nncf.quantize` method on a Keras/TF model

**Reply by MaximProshin** [reply, marked as answer]
@jmmanso , for PTQ we recommend OV path, i.e. TF->OV->NNCF->int8_OV, in that case there wouldn't be any dependency on TF version in NNCF (see more in https://docs.openvino.ai/2024/openvino-workflow/model-optimization-guide/quantizing-models-post-training/basic-quantization-flow.html). Please consider that option. 
Anyway, we also consider to support TF 2.18, that was just released, in future NNCF versions but I don't have clear plans yet.

**Reply by dbrandonk** [reply]
Hey Maxim, has any plans been made yet to support TF 2.18.0+/ Keras 3? From what I can tell no library exists to support Keras 3 QAT! Or training time pruning for that matter!

**Comment by MaximProshin** [comment]
@jmmanso , we tried to enable it but faced with some issues with Keras 3 which force us to re-write existing functionality and it will take quite some time. Currently it's not in our priorities list. cc @alexsu52 @andrey-churkin 
@EfimovIlia , please take a look at this request from the Product team standpoint and let's discuss the priority after that. Internal ticket number is 155857 for your reference.

**Reply by dbrandonk** [reply]
Got it, thank you for the response. I'll probably keep checking in until a solution somewhere exists haha.
